src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/JavaScriptTests.cs

```diff
@@ -274,28 +279,36 @@ public static IEnumerable<object> ToEnumerable(this JSObject iterrator)
     },
 };
 ");
-            return (JSObject)Runtime.GetGlobalObject(prefix);
+            return (JSObject)factory.Call();
         }
 
         [Fact]
         public static void AddEventListenerWorks () {
-            var temp = new bool[1];
-            var obj = SetupListenerTest("addEventListenerWorks");
-            obj.AddEventListener("test", () => {
-                temp[0] = true;
+            var temp = new bool[2];
```

Comment:
I don't think this test case should be mixing in multiple events and GC. It should remain a simple test that just covers the most obvious thing, and we have a separate test for more complex scenarios. As-is if the basics break this test isn't as useful for diagnostics since so many things are happening at once.
(I'm fine with new tests you add being complex, I just don't want to see all our simple tests become complex ones)
Also note that the relatively sparse design of the original test doesn't scale up - if this breaks the error messages won't be very helpful, especially since wasm stacktraces don't have line numbers. We'd want to make it emit more detailed messages upon failure if we've removed the simple tests.

Reply:
Good feedback, I separated the test cases.